jerry-core/parser/js/js-parser-expr.c

```diff
@@ -1189,30 +1190,93 @@ parser_parse_template_literal (parser_context_t *context_p) /**< context */
 
     if (context_p->last_cbc_opcode == CBC_PUSH_LITERAL)
     {
-      context_p->last_cbc_opcode = CBC_ADD_RIGHT_LITERAL;
+      context_p->last_cbc_opcode = PARSER_TO_EXT_OPCODE (CBC_EXT_STRING_CONCAT_RIGHT_LITERAL);
     }
     else
     {
-      parser_emit_cbc (context_p, CBC_ADD);
+      parser_emit_cbc_ext (context_p, CBC_EXT_STRING_CONCAT);
     }
 
     context_p->source_p--;
     context_p->column--;
-    lexer_parse_string (context_p);
+    lexer_parse_string (context_p, LEXER_STRING_NO_OPTS);
 
     if (context_p->token.lit_location.length > 0)
     {
       lexer_construct_literal_object (context_p,
                                       &context_p->token.lit_location,
                                       context_p->token.lit_location.type);
 
-      parser_emit_cbc_literal_from_token (context_p, CBC_ADD_RIGHT_LITERAL);
+      parser_emit_cbc_ext_literal_from_token (context_p, CBC_EXT_STRING_CONCAT_RIGHT_LITERAL);
     }
   }
-
-  return;
 } /* parser_parse_template_literal */
 
+/**
+ * Parse tagged template literal.
+ */
+static size_t
+parser_parse_tagged_template_literal (parser_context_t *context_p) /**< context */
+{
+  JERRY_ASSERT (context_p->token.type == LEXER_TEMPLATE_LITERAL);
+
+  uint32_t call_arguments = 0;
+  ecma_collection_t *collection_p;
+
+  if (context_p->tagged_template_literal_cp == JMEM_CP_NULL)
+  {
+    collection_p = ecma_new_collection ();
+    ECMA_SET_INTERNAL_VALUE_POINTER (context_p->tagged_template_literal_cp, collection_p);
+  }
+  else
+  {
+    collection_p = ECMA_GET_INTERNAL_VALUE_POINTER (ecma_collection_t, context_p->tagged_template_literal_cp);
+    if (collection_p->item_count > CBC_MAXIMUM_BYTE_VALUE)
+    {
+      parser_raise_error (context_p, PARSER_ERR_ARGUMENT_LIMIT_REACHED);
+    }
+  }
+
+  const uint32_t tagged_id = collection_p->item_count;
+  uint32_t prop_idx = 0;
+  ecma_object_t *raw_strings_p;
+  ecma_object_t *template_obj_p = parser_new_tagged_template_literal (&raw_strings_p);
```

Comment:
Do we need this helper? Looks simple, and used only once.

Reply:
I'd prefer these helpers functions. I think it helps to separate the runtime functionality from the parse time also the whole property management is performed here.